content/user/apps/DiscoverVulns.md

```diff
@@ -4,23 +4,77 @@ description: "Overview of discovering vulnerabilities"
+
+The grid of discovered vulnerabilities provides extremely detailed information on each one, including: 
 
 * Type of vulnerability
 * Tutorial on how to fix it
 * Line of code
 * Ability to replay the attack
 
-## How Vulnerabilities Work
-If a vulnerability is reported and Contrast has never seen it before, Contrast creates a new vulnerability. However, if that vulnerability already exists, Contrast updates the existing entry, issue count and number of days since it was last detected.
+## Export Findings
+
+Export details on vulnerability findings from the **Vulnerabilities** page, a vulnerability's **Overview** tab, or an application or server's **Vulnerabilities** tab. Click the **Export** icon to choose either CSV or XML formats for the grouping of vulnerabilities that you want to include in the report.
 
-<a href="assets/images/Application-vulnerabilities-tab.png" rel="lightbox" title="Vulnerability Details"><img class="thumbnail" src="assets/images/Application-vulnerabilities-tab.png"/></a>
+<a href="assets/images/Vuln-export-options.png" rel="lightbox" title="Vulnerability export options"><img class="thumbnail" src="assets/images/Vuln-export-options.png"/></a>
 
->**Example:** A vulnerability was reported to Contrast five times for one server. Instead of showing up as five vulnerabilities, Contrast updates this entry and increments the count. As Contrast continues to see the same findings, the count goes up. If you dive into the Notes tab within this vulnerability, you notice a list of the servers in which this vulnerability was found.
+The exported file contains the following data fields for each vulnerability:
+
+* Vulnerability Name
+* Vulnerability ID
+* Category
+* Rule Name
+* Severity
+* Status
+* Number of Events
+* First Seen
+* Last Seen
+* Application Name
+* Application ID
+* Application Code
+* Application Version
+* CWE ID
+* Request Method
+* Request Port
+* Request Protocol
+* Request Version
+* Request URI
+* Request Qs
+* Request Body
+
+### Custom reports
+
+For users looking to craft custom software composition analysis reports about their applications, the vulnerability export feature might not provide sufficient information; however, Contrast offers a rich Application API which gives you access to Contrast vulnerability data. Reference the **Contrast RESTful API documentation > Application Trace Filtering > /ng/{orgUuid}/traces/{appId}/filter** section for instructions on using the Application API. 
+
+You may also explore additional details on your vulnerabilities using a manual method.
+
+> **Example:** This cURL request retrieves a list of vulnerabilities that also shows a list of the applications in which each vulnerability was found. The jq tool formats the data as CSV for use in a custom report.
+```
+curl -H "Authorization: $(echo -n $username:$servicekey
```

Comment:
It looks like something removed the `|` characters. That and the line breaks makes this command syntactically incorrect. It should be

```
curl \
    -H "Authorization: $(echo -n $username:$servicekey | base64)" \
    -H "API-Key: $apikey" \
    https://app.contrastsecurity.com/Contrast/api/ng/$orgid/orgtraces/filter?expand=request | \
    jq -r '.traces[] | {uuid: .uuid, protocol: .request.protocol} | [.uuid, .protocol] | @csv'
```

Reply:
Copied this to the file.